### PR TITLE
DVOP-1939 adds slack notifications

### PR DESCRIPTION
**Pull request by jamesgeddes**
ensures that a slack notification is sent any time that a logging.error is triggered. not the most dry solution so could be improved in the future with a global logging.error listener.

### CODE REVIEW

File: source/webhook-trigger.py [hunk 1]
```diff
@@ -12,30 +12,53 @@
...
-        + " but Python %s.%s or later is required.\n" % MIN_PYTHON
-    )
-
 
 VARIABLES = {
     "CONCOURSE_URL": "The URL through which we'll force resource checks",
     "SLACK_WEBHOOK_URL": "The URL of the Slack webhook we will use to publish messages",
 }
 
+MIN_PYTHON = (3, 7)
```
Comment (Jata2k) [inline source/webhook-trigger.py:21]: I'd probably leave this and the version check statement at the top and certainly together

Comment (jamesgeddes) [reply source/webhook-trigger.py:21]: It would appear that I am declaring it twice. Typo. I'll remove.

Comment (jamesgeddes) [reply source/webhook-trigger.py:21]: The `send_slack_error_message` function needs to be defined for the py version check to use it. I have therefore moved the version check to the bottom outside of all the function definitions.

**Review by sienkin — APPROVED** [review]

**Comment by jamesgeddes** [comment]
Thanks @sienkin 😄

File: source/webhook-trigger.py [hunk 2]
```diff
@@ -97,26 +110,17 @@ def verify_environment(variables):
...
-        values["status_code"] = status_code
-
-    slack_webhook_url = os.getenv("SLACK_WEBHOOK_URL")
-
-    loader = jinja2.FileSystemLoader("templates")
-    env = jinja2.Environment(loader=loader)
-    env.filters["jsonify"] = json.dumps
+if sys.version_info < MIN_PYTHON:
+    error_message = "Found Python " + str(sys.version_info) + " but Python %s.%s or later is required. " \
+                                                              "Unable to find towel. Panicking.\n" % MIN_PYTHON
```
Comment (Jata2k) [inline source/webhook-trigger.py:120]: As discussed we should probably keep the error messages concise

File: source/webhook-trigger.py [hunk 3]
```diff
@@ -12,30 +12,53 @@
...
+    send_slack_error_message(error_message, None)
+    sys.exit(error_message)
+
 
 logging.basicConfig()
 logging.getLogger().setLevel(logging.INFO)
 
 
 def exception_message(origin_message, exception_object):
```
[the remark this answers is not in the document]
Comment (jamesgeddes) [reply source/webhook-trigger.py:45]: `send_slack_error_message` sends a slack message on lines [38 to 42](https://github.com/companieshouse/ci-webhook-trigger/blob/28f5c5a1e2822b8bd29d199a5c42af8b01cc0cfb/source/webhook-trigger.py#L38-L42). Have renamed `error_message` to `build_exception_message`.

File: source/webhook-trigger.py (hunk 2 above)
Comment (jamesgeddes) [reply source/webhook-trigger.py:120]: Have changed to `"Found Python " + str(sys.version_info) + " but need Python %s.%s or later." % MIN_PYTHON`

File: source/webhook-trigger.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamesgeddes) [reply source/webhook-trigger.py:45]: #7 now available. Let's continue there.

**Pull request merged**